Add support for other Body types

karaggeorge opened the pull request:
The current ruby client only supports a request body of type: `STRING`, `REGEX`, `XPATH`, `PARAMETERS` and `BINARY`. 

The server supports a couple more. My team is using the client and we are trying to use `JSON` to set our expectations because using `exact` forces us to pass the arguments in the exact same order as the expectation. 

This PR adds support on the client for these extra body types that are supported on the server: `JSON`, `JSON_SCHEMA` and `XML_SCHEMA`. 

This PR also refactors the integration test so the context and examples can be reused and adds tests to ensure the interaction with the new body types works. The tests all pass.

I also noticed that the coverage is not working in this project. It detects 0 lines, so I couldn't explore if the test coverage went up or down. However, every line of code added was tested.

karaggeorge commented:
Travis-CI build fails because of linting errors in parts of the code that this PR didn't touch. I think the master branch fails to build as well

jamesdbloom commented:
I am planning on completely re-writting the ruby client.  I intend to do that shortly as soon as I have completed exposing the swagger file for the REST API.

karaggeorge commented:
@jamesdbloom do you think this PR can get merged until that happens? It's a very basic change, just adding support for JSON and a few other body types. All it changed was adding a few extra keys to the BodyTypes class and the matching methods. 

It has also been tested that the integration with the server works.

jamesdbloom commented:
It won't be possible to merge the PR until the build is fixed, please take a look at the failures on Travis, as shown below.

karaggeorge commented:
@jamesdbloom I did, and all the tests pass. Travis is failing because of linting errors that existed in the master branch. The master branch of the repo fails on travis-ci as well...

karaggeorge commented:
@jamesdbloom I went ahead and fixed all the unrelated linting errors that this repo had, updated outdated dependencies, and ensured all the tests pass. However, the build failed again because there was no .travis.yml file to specify the build. So, I went ahead and wrote a .travis.yml file that checks the linting errors by runnint `rubocop` and also runs the tests by executing the `build.sh` script.

Linting and tests now pass on Travis-CI.

Sidenote: It's a bit counterintuitive to expect contribution PRs to pass the build before merging, when even the master branch fails and/or there is no build configuration in place for the tests to even run

chrisbutler commented:
@jamesdbloom any chance of re-considering this for a merge? looks like @karaggeorge has met all of your criteria
